README.md

```diff
@@ -80,7 +80,74 @@ These are the cli options exposed to `pytest` by the plugin.
 | `--snapshot-warn-unused`       | Prints a warning on unused snapshots rather than fail the test suite.               | `False`                                          |
 | `--snapshot-default-extension` | Use to change the default snapshot extension class.                                 | `syrupy.extensions.amber.AmberSnapshotExtension` |
 
-### Built-In Extensions
+### Assertion Options
+
+These are the options available on the `snapshot` assertion fixture.
+Use of these options are one shot and do not persist across assertions.
+For more persitent options see the [advanced usage](#advanced-usage).
+
+#### `matcher`
+
+This allows you to match on a property path and value to control how specific object shapes are serialized.
+
+The matcher is a function that takes two keyword arguments:
+
+| Argument | Description                                                                   |
+| -------- | ----------------------------------------------------------------------------- |
+| `data`   | Current serializable value being matched on                                   |
+| `path`   | Ordered path traversed to the current value e.g. `(("a", dict), ("b", dict))` |
+
+**NOTE:** Do not mutate the value received as it could cause unintended side effects.
+
+##### Built-In Matchers
+
+Syrupy comes with a few built-in matcher presets that can be used to make easy work of using property matchers.
+
+###### `path_type(mapping=None, *, types=(), strict=True)`
+
+Easy way to build a matcher that uses the path and value type to replace serialized data.
+When strict, this will raise a `ValueError` if the types specified are not matched.
+
+| Argument  | Description                                                                                                                        |
+| --------- | ---------------------------------------------------------------------------------------------------------------------------------- |
+| `mapping` | Dict of path string to tuples of class types                                                                                       |
+| `types`   | Tuple of class types used if none of the path strings from the mapping are matched                                                 |
```

Comment:
Does it only work on class types? What about primitive types?

Reply:
Primitives work as well e.g. https://github.com/tophat/syrupy/blob/6fa24059755cc29524ce223993bd3e6392ecb6d5/tests/test_matchers.py#L28-L33